dali/operators/image/crop/bbox_crop.cc

```diff
@@ -507,12 +561,33 @@ class RandomBBoxCropImpl : public OpImplBase<CPUBackend> {
       extent = max_extent * extent / new_max_extent;
   }
 
-  const ProspectiveCrop FindProspectiveCrop(span<const Box<ndim, float>> bounding_boxes,
-                                            span<const int> labels,
-                                            int sample) {
+  ProspectiveCrop FindProspectiveCrop(span<const Box<ndim, float>> bounding_boxes,
+                                      span<const int> labels, int sample) {
+    ProspectiveCrop crop;
+    int count = 0;
+    while (!crop.success && count < total_num_attempts_) {
+      auto &rng = rngs_[sample];
+      std::uniform_int_distribution<> idx_dist(0, sample_options_.size() - 1);
+      SampleOption option = sample_options_[idx_dist(rng)];
+
+      crop = FindProspectiveCrop(bounding_boxes, labels, sample, option);
+      count += num_attempts_;
+    }
+
+    if (!crop.success) {
+      DALI_FAIL(
```

Comment:
I don't like this approach. It could happen once, at the very end of several hours of training and then someone would be very pissed-off. I would make this fail optional, and maybe just skip cropping as a default behavior.